js/apps/admin-ui/src/user/UserProfileFields.tsx

```diff
@@ -200,8 +206,12 @@ function determineInputType(
     return DEFAULT_INPUT_TYPE;
   }
 
-  // An attribute with multiple values is always multi-valued, even if an input type is provided.
-  if (Array.isArray(value) && value.length > 1) {
+  // For a non-multi value allowed input type, check if multiple values are provided, if so fall back to multi-input.
+  if (
```

Comment:
I think we should improve the `determineInputType()` function a little, so that there is a bit more explanation behind the choices being made here:

```ts
function determineInputType(
  attribute: UserProfileAttributeMetadata,
  value: string | string[],
): InputType {
  // Always treat the root attributes as a text field.
  if (isRootAttribute(attribute.name)) {
    return "text";
  }

  const inputType = attribute.annotations?.inputType;

  // If the attribute has no valid input type, fall back to a default input type.
  // Depending on the length of the value, we either use a 'multi-input' or a 'text' input type so all values are always visible.
  if (!isValidInputType(inputType)) {
    return value.length > 1 ? "multi-input" : "text";
  }

  // If the input type is multi-valued, we don't have to do any further checks, as we know all values will always show up.
  if (MULTI_VALUED_INPUT_TYPES.includes(inputType)) {
    return inputType;
  }

  // An attribute with multiple values is always as a 'multi-input', even if a singular input type is provided.
  // This is done so that the user can edit the attribute without accidentally truncating the other values that would otherwise be hidden.
  if (Array.isArray(value) && value.length > 1) {
    return "multi-input";
  }

  return inputType;
}
```

WDYT?